src/sdl_ui.cpp

```diff
@@ -171,12 +171,17 @@ SdlUi::SdlUi(long width, long height, bool fs_flag) :
 #endif
 
 #ifdef HAVE_SDL_MIXER
-	audio_.reset(new SdlAudio());
+	if (!Player::no_audio_flag) {
+		audio_.reset(new SdlAudio());
+		return;
+	}
 #elif defined(HAVE_OPENAL)
-	audio_.reset(new ALAudio());
-#else
-	audio_.reset(new EmptyAudio());
+	if (!Player::no_audio_flag) {
+		audio_.reset(new ALAudio());
+		return;
+	}
 #endif
+	audio_.reset(new EmptyAudio());
 }
 
 SdlUi::~SdlUi() {
```

Comment:
No, to check for that you would need to initialize SDL with audio and fallback to no-audio, if it fails.